src/api-v3/paths/payments.ts

```diff
@@ -11,6 +15,42 @@ import { ERRORS } from "../../errors";
 const config = getConfig();
 
 export default function(api: RippleAPI, log: Function): Operations {
+  async function put(req: Request, res: ValidatableResponse, _next: NextFunction): Promise<void> {
+    let signedTransaction = req.body.signedTransaction
+    if (signedTransaction === undefined) {
+      // Do not parse X-address, if present. This endpoint takes transactions fully-formed.
+      // It only signs and submits them. To prepare transactions (converting X-addresses, if necessary),
+      // use GET /v3/preparations/payments.
+
+      if (!api.isValidAddress(req.body.Account)) {
+        throw new Error('Invalid `Account`');
+      }
+
+      if (!config.accounts || !config.accounts[req.body.Account]) {
+        throw ERRORS.ACCOUNT_NOT_CONFIGURED;
+      }
+
+      const accountWithSecret = config.accounts[req.body.Account];
+
+      // Require valid Bearer Token
+      let reqHasValidBearerToken = false;
+      if (!accountWithSecret || 'Bearer ' + accountWithSecret.apiKey != req.headers.authorization) {
+        log(`[401] does not match apiKey, authorization: ${req.headers.authorization}`);
+        throw ERRORS.INVALID_BEARER_TOKEN;
+      } else {
+        reqHasValidBearerToken = true;
+      }
+
+      if (reqHasValidBearerToken) { // In the future, apply velocity limits here
+        signedTransaction = api.sign(JSON.stringify(req.body), accountWithSecret.secret).signedTransaction;
+      }
+    }
+    const result = await api.submit(signedTransaction);
+    delete result.resultCode;    // (use `engine_result` instead)
+    delete result.resultMessage; // (use `engine_result_message` instead)
```

Comment:
What do you mean `use 'engine_result_message' instead`?

Reply:
`result.resultMessage` and `result.engine_result_message` are always the same, but `engine_result_message` is the preferred one since it is used by the rippled API and `resultMessage` will be removed in ripple-lib 2.0.

Reply:
Got it, so this is a TODO? Can we add `TODO` to the comment?

Reply:
No, the only thing to do is to remove the property in ripple-lib, which is a breaking change, so we'll make it in ripple-lib 2.0.

Reply:
Oh, I see. The `result` object has both of those properties, but you're removing the deprecated properties from the object.